cookbooks/fb_consul/recipes/default.rb

```diff
@@ -0,0 +1,157 @@
+#
+# Cookbook:: fb_consul
+# Recipe:: default
+#
+# Copyright (c) 2020-present, Vicarious, Inc.
+# All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
```

Comment:
make it fail on unsupported platforms like we do in other cookbooks

Reply:
Sure. Actually I've been thinking about that.

The `fail`s we used to do were originally for "this can never possible work on this platform you have done something wrong" - like including fb_systemd on non-systemd platforms (macosx) for example. But in the last few contributions we ended up making the `fail` be the inverse of the `supports` in the metadata... and I can see an argument for that... but I think since that data is already there perhaps a better way is keeping the original meaning. So for example here I"d do `fail unless node.linux?` - I haven't tested this on the RH-like distros, but it's not meaningless on them. What do you think?

Reply:
This isn't meaningless on RH, but it's not been tested there and likely won't work properly. I'd rather it fail early, so one can check what's going on and test and fix it, rather than be surprised.